content/en/docs/concepts/security/security-checklist.md

```diff
@@ -0,0 +1,333 @@
+---
+reviewers:
+title: Security Checklist
+description: >
+  Baseline checklist for ensuring security in Kubernetes clusters.
+content_type: concept
+---
+
+<!-- overview -->
+
+This checklist aims at providing a basic list of guidance with links to more
+comprehensive documentation on each topic. It does not claim to be exhaustive
+and is meant to evolve.
+
+The order of topics does not reflect an order of priority.
+
+{{< caution >}}
+Checklists are **not** sufficient for attaining a good security posture on their
+own. A good security posture requires constant attention and improvement, but a
+checklist can be the first step on the never-ending journey towards security
+preparedness.  Some of the recommendations in this checklist may be too
+restrictive or too lax for your specific security needs. Since Kubernetes
+security is not "one size fits all", each category of checklist items should be
+evaluated on their merits.
+{{< /caution >}}
+
+<!-- body -->
+
+## Authentication & Authorization
+
+- [ ] `system:masters` group is not used for user or component authentication after bootstrapping.
+- [ ] The kube-controller-manager is running with `--use-service-account-credentials`
+  enabled.
+- [ ] The root certificate is protected (either an offline CA, or a managed
+  online CA with effective access controls).
+- [ ] Intermediate and leaf certificates have an expiry date no more than 3
+  years in the future.
+- [ ] A process exists for periodic access review, and reviews occur no more
+  than 24 months apart.
+
+See the [Role Based Access Control Good Practices](/docs/concepts/security/rbac-good-practices/)
+for guidance related to authentication and authorization.
+
+After bootstrapping, neither users nor components should authenticate to the
+Kubernetes API as `system:masters`. Similarly, running all of
+kube-controller-manager as `system:masters` should be avoided. In fact,
+`system:masters` should only be used as a break-glass mechanism, as opposed to
+an admin user.
+
+## Network security
+
+- [ ] CNI plugins in-use supports network policies.
+- [ ] Network policies to filter how workloads and services can communicate
+  inside of the cluster are enabled.
+- [ ] Use of a service mesh to encrypt all communications inside of the cluster.
```

Comment:
Do we want to scope this checklist to Kubernetes and not third-party tools
Which is difficult because there are third-party things Kubernetes needs like a CNI plugin and CRI-compliant container runtime
I can be persuaded either way

Reply:
We had many discussions about third-party tools (for example here https://github.com/kubernetes/website/pull/33992#pullrequestreview-993832115) and I think the conclusion was that it was a bit impossible to do so on a doc page. The only thing we could do is to link to CNCF pages, blogs or landscape parts that contain projects amongst the community for these topics!

I think if we did add many links to third-party stuff, it will take a majority of all the hyperlinks on the page, look at what was done here https://github.com/Vinum-Security/kubernetes-security-checklist. The work is truly great but there are a lot of links to third-party tools (which make sense in the context of Kubernetes security I guess).